web/src/components/filings-calendar/tax-access/TaxAccess.tsx

```diff
@@ -0,0 +1,53 @@
+import { Content } from "@/components/Content";
+import { TaxAccessStepOne } from "@/components/filings-calendar/tax-access/TaxAccessStepOne";
+import { TaxAccessStepTwo } from "@/components/filings-calendar/tax-access/TaxAccessStepTwo";
+import { Heading } from "@/components/njwds-extended/Heading";
+import { useConfig } from "@/lib/data-hooks/useConfig";
+import { useUserData } from "@/lib/data-hooks/useUserData";
+import analytics from "@/lib/utils/analytics";
+import { useMountEffectWhenDefined } from "@/lib/utils/helpers";
+import { ReactElement, useEffect, useState } from "react";
+
+interface Props {
+  onSuccess: () => void;
+}
+
+export const TaxAccess = (props: Props): ReactElement => {
+  const [isStepOne, setIsStepOne] = useState<boolean>(true);
+  const [hadLegalStructureOnMount, setHadLegalStructureOnMount] = useState<boolean>(false);
+  const { business } = useUserData();
+  const { Config } = useConfig();
+
+  useMountEffectWhenDefined((): void => {
+    if (business?.profileData.legalStructureId) {
+      setIsStepOne(false);
+      setHadLegalStructureOnMount(true);
+    }
+  }, business);
+
+  useEffect(() => {
+    analytics.event.tax_calendar.arrive.arrive_calendar_access_v2();
```

Comment:
pebble - based on how useMountEffectWhenDefined is triggered (when business is available), and fired once, we could move this analytics function inside useMountEffectWhenDefined and get rid of this useEffect.